src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1295,12 +1295,11 @@ private function wrapServiceConditionals($value, $code)
      */
     private function getServiceConditionals($value)
     {
-        if (!$services = ContainerBuilder::getServiceConditionals($value)) {
-            return null;
-        }
-
         $conditions = array();
-        foreach ($services as $service) {
+        foreach (ContainerBuilder::getInitializedConditionals($value) as $service) {
+            $conditions[] = sprintf("isset(\$this->services['%s'])", $service);
```

Comment:
this needs to be updated based on service visibility when merging into master, right ? Does you testing covers this case for both private and public services ?

Reply:
done - and allowed me to spot an issue: now, the service graph is aware of those "weak" edges